test/common/run-tests

```diff
@@ -215,14 +215,12 @@ def run(opts, image):
     # If file from `test-dir` was changed in the PR, all tests from it are considered affected
     changed_tests = []
     test_files = glob.glob(os.path.join(opts.test_dir, "check-*"))
-    if os.path.exists(os.path.join(os.path.dirname(testvm.TEST_DIR), ".git")):
+    if opts.base:
         # Detect affected tests from changed test files
-        cmd = ["git", "diff", "--name-only", "origin/" + opts.base, opts.test_dir]
-        r = subprocess.run(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
-        if r.returncode == 0:
-            # Never consider 'test/verify/check-example' to be affected - our tests for tests count on that
-            # This file provides only examples, there is no place for it being flaky, no need to retry
-            changed_tests = [test.decode("utf-8") for test in r.stdout.strip().splitlines() if not test.endswith(b"check-example")]
+        changed_files = subprocess.check_output(["git", "diff", "--name-only", "origin/" + opts.base, opts.test_dir])
```

Comment:
on lines 238-240 we have the same pattern, want to change that one as well to `check_output`?

Reply:
I confirmed that non-existing paths (`pkg/` only exist in cockpit) does not make it fail, so I'll do that.

Reply:
Done. I renamed the variable to `diff_out`, as otherwise with that schema we would have `changed_pkg` and `changed_pkgs`  below, which is confusing.